### DISCUSSION TITLE
rust toolchain override causes rust-analyzer language server to exit.

### DISCUSSION THREAD

**Discussion by jchidley** [helix-editor/helix#11613] [General] [answered]
[rust toolchain override](https://rust-lang.github.io/rustup/overrides.html) causes rust-analyzer Language server to exit.

`rustup toolchain list`:

```
stable-x86_64-pc-windows-msvc (default)
1.80-x86_64-pc-windows-msvc (override)
```

`helix.log`:

```
2024-08-31T14:20:03.024 helix_lsp::transport [ERROR] rust-analyzer err <- "error: Unknown binary 'rust-analyzer.exe' in official toolchain '1.80-x86_64-pc-windows-msvc'.\n"
2024-08-31T14:20:03.074 helix_lsp::transport [ERROR] rust-analyzer err: <- StreamClosed
2024-08-31T14:20:03.074 helix_lsp [ERROR] failed to initialize language server: server closed the stream
```

`rust-toolchain.toml`:

```
[toolchain]
channel = "1.80"
components = [ "rust-src", "rustfmt", "llvm-tools" ]
targets = [
    "thumbv7em-none-eabi",
    "thumbv7m-none-eabi",
    "thumbv6m-none-eabi",
    "thumbv7em-none-eabihf",
    "thumbv8m.main-none-eabihf",
    "riscv32imac-unknown-none-elf",
    "wasm32-unknown-unknown",
]
```

VS Code works fine with this configuration, helix doesn't.

## Mitigation

Start helix from a directory where the override isn't in effect to edit the desired file.

**Comment by kirawi** [comment, marked as answer]
You have to install rust-analyzer for the new toolchain.

**Comment by jchidley** [comment]
Thanks. I spent way too long just getting working out a mitigation.  Why does this work out of the box in VS Code but require extra steps in helix?

Jack

**Reply by kirawi** [reply]
VSCode manages its own rust-analyzer version. You could also replicate this in Helix by managing rust-analyzer with a separate package manager rather than rustup.
